src/domains/ecommerce-app/api/ecommerce-checkout/v1/_validate_transactions_jwt_token.tpl

```diff
@@ -1,10 +1,14 @@
 <policies>
     <inbound>
         <base />
-        <set-header name="x-transaction-id" exists-action="delete" />
         <set-variable name="requestTransactionId" value="@{
-            return context.Request.MatchedParameters["transactionId"];
-        }" />
+                    var transactionId = context.Request.MatchedParameters.GetValueOrDefault("transactionId","");
+                    if(transactionId == ""){
+                        transactionId = context.Request.Headers.GetValueOrDefault("x-transaction-id","");
+                    }
+                    return transactionId;
+            }" />
+        <set-header name="x-transaction-id" exists-action="delete" />
```

Comment:
@infantesimone  the header delete should be unnecessary since then an override is done after the JWT validation.
`<set-header name="x-transaction-id" exists-action="override">`
Do you confirm ?

Reply:
ping!  🙂 @infantesimone

Reply:
it is a check that makes the policy more secure, imagine a bug in the policy section which for some conditions does not value the header. At this point the header would not arrive at the service. Without this check, an attacker could inject a value of the a header from outside